addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,326 @@
+        if req['event'].wait(58):
+            req['event'].clear()
+            return_value = req['queue'].copy()
+            return return_value
+
+    @http.route('/hw_drivers/box/connect', type='http', auth='none', cors='*', csrf=False)
+    def connect_box(self, token):
+        server = get_odoo_server_url()
+        if server:
+            f = open('/var/www/False.jpg','rb')
+            return f.read()
+        else:
+            iotname = ''
+            token = b64decode(token).decode('utf-8')
+            url = token.split('|')[0]
+            token = token.split('|')[1]
+            reboot = 'noreboot'
+            path = os.getenv("HOME") + '/odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh'
+            subprocess_call([path, url, iotname, token, reboot])
+            m.send_alldevices()
+            f = open('/var/www/True.jpg','rb')
+            return f.read()
 
 #----------------------------------------------------------
-# Bluetooth
+# Drivers
 #----------------------------------------------------------
-class GattBtManager(gatt.DeviceManager):
-
-    def device_discovered(self, device):
-        # TODO: need some kind of updated_devices mechanism or not?
-        path = "bt_%s" % (device.mac_address,)
-        if path not in drivers:
-            for driverclass in btdrivers:
-                d = driverclass(device = device, manager=self)
-                if d.supported():
-                    drivers[path] = d
-                    d.connect()
-                    send_iot_box_device(False)
 
+drivers = []
+bt_devices = {}
+iot_devices = {}
 
-class BtManager(Thread):
-    gatt_manager = False
-
-    def run(self):
-        dm = GattBtManager(adapter_name='hci0')
-        self.gatt_manager = dm
-        dm.start_discovery()
-        dm.run()
-
-#----------------------------------------------------------
-# Bluetooth drivers
-#----------------------------------------------------------
-btdrivers = []
-
-class BtMetaClass(type):
+class MetaClass(type):
     def __new__(cls, clsname, bases, attrs):
-        newclass = super(BtMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        btdrivers.append(newclass)
+        newclass = super(MetaClass, cls).__new__(cls, clsname, bases, attrs)
+        drivers.append(newclass)
         return newclass
 
-
-class BtDriver(Driver, metaclass=BtMetaClass):
-
+class Driver(Thread, metaclass=MetaClass):
+    connection_type = ""
 
     def __init__(self, device, manager):
-        super(BtDriver, self).__init__()
+        super(Driver, self).__init__()
         self.dev = device
         self.manager = manager
-        self.value = ''
+        self.value = ""
+        self.data = ""
         self.gatt_device = False
 
-    def disconnect(self):
-        path = "bt_%s" % (self.dev.mac_address,)
-        del drivers[path]
-
     def get_name(self):
-        return self.dev.alias()
+        return ''
+
+    def get_identifier(self):
```

Comment:
you should maybe explore `@property`
you would be able to use driver.identifier instead of driver.get_identifier()